nat.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/sh
+
+
+#
+#--------------------------------- constants
+fn_check_dependencies() {
+	local __cmd=${1}
+	local __path=$(which ${__cmd})
+
+	if [ $? -ne 0 ]; then
+		echo "****************** !! \${__cmd}\' IS NOT AVAILABLE !! ******************"
+		exit 1
+	fi
+
+	echo ${__path}
+}
+
+#
+#--------------------------------- constants
+AWK=$(fn_check_dependencies awk)
+ECHO=$(fn_check_dependencies echo)
+DEPMOD=$(fn_check_dependencies depmod)
+GREP=$(fn_check_dependencies grep)
+IP=$(fn_check_dependencies ip)
+IPTABLES=$(fn_check_dependencies iptables)
+MODPROBE=$(fn_check_dependencies modprobe)
+NETSTAT=$(fn_check_dependencies netstat)
+
+#
+#--------------------------------- functions
+fn_get_wan_iface() {
+	${ECHO} $(${IP} route show | ${GREP} default | ${AWK} '{print $5}')
+}
+
+fn_get_lan_iface() {
+	local _wan_ip=$(fn_get_wan_iface)
+	${ECHO} $(${NETSTAT} -i | ${GREP} -ve lo -ve Iface -ve Kernel -ve ${_wan_ip} | ${AWK} '{print $1}')
```

Comment:
This will return "" <empty string> when there is only one interface. Check the issue #1